mds/api/client.py

```diff
@@ -120,9 +120,7 @@ def __next_url(page):
 
                 if __has_data(this_page):
                     results[provider].append(this_page)
-                    next_url = __next_url(this_page)
-                else:
-                    break
+                next_url = __next_url(this_page)
```

Comment:
An additional change that I'll request is to `break` if `next_url == this_page["links"]["last"]`. Can you please add this as another `if` block before this line?

Go ahead and create a helper `__last_url(page)` if you want, like `__next_url(page)`.

Reply:
hm, im a little confused by this proposed change.

after reading over our discussion here #57 , i'm now convinced that the following should be the case for `next_url`:
>I would argue it is not in spec for a response to contain a next_url pointing to data outside of the requested window.
in other words, next_url will be null when all data requested by the client has been paginated through _or_ the client requested a window where there simply doesnt exist data to page through

therefore, the `__next_url(this_page)` call should take care of the case for when we're done (ie. when next_url is empty). the MDS spec says: 

>At a minimum, paginated payloads must include a next key, which must be set to null to indicate the last page of data.

so i don't think we need to check if next url = last url

@thekaveman what do you think?

Reply:
the way i model this `links` block to work is, as according to the specs: https://jsonapi.org/format/#fetching-pagination

that it's essentially like a **doubly linked list** -- you have first, last, prev, and next pointers. when you're at the last page, next should point to null.

Reply:
@niketanpatel you are right, I was thinking of before `next` was required... sorry, ignore that request.